[1LP][RFR] Replace usage of entity.check() with entity.ensure_checked. Part 1.

prichard77 opened the pull request:
## Purpose or Intent

- __Updating tests and framework__ to use entity ensure_checked() instead of entity.check(). ensure_checked() is an enhancement to check() functionality that accounts for the checked state of an entity when selecting it.

### PRT Run

{{ pytest: --use-provider rhv42 --use-provider rhos13 --long-running cfme/tests/cloud_infra_common/test_policy_simulation.py::test_policy_simulation_ui cfme/tests/cloud/test_instance_power_control.py::test_power_on_or_off_multiple cfme/tests/cloud_infra_common/test_vm_instance_analysis.py::test_ssa_multiple_vms cfme/tests/infrastructure/test_host.py::test_multiple_host_good_creds }}

dajoRH commented:
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

prichard77 commented on cfme/cloud/instance/__init__.py:
```diff
@@ -318,7 +318,7 @@ def power_control_from_cfme(self, *args, **kwargs):
                     'Failed to find instance in table: {}'.format(self.name)
                 )
-            row.check()
+            row.ensure_checked()
```
I have only been able to test this by changing the surrounding code so line 321 would be executed. IT tested fine, but there is not a specific test in PRT that covers this update.

prichard77 commented on cfme/cloud/instance/__init__.py:
```diff
@@ -643,7 +643,8 @@ def prerequisite(self):
         # click the checkbox of every object in the filtered collection
         for entity in self.obj.all():
-            self.prerequisite_view.entities.get_entity(name=entity.name, surf_pages=True).check()
+            self.prerequisite_view.entities.get_entity(name=entity.name,
```
This is covered by cfme/tests/cloud_infra_common/test_policy_simulation.py::test_policy_simulation_ui and will be added to PRT.

prichard77 commented on cfme/cloud/provider/__init__.py:
```diff
@@ -266,7 +266,8 @@ class Edit(CFMENavigateStep):
     def step(self, *args, **kwargs):
-        self.prerequisite_view.entities.get_entity(name=self.obj.name, surf_pages=True).check()
+        self.prerequisite_view.entities.get_entity(name=self.obj.name,
+                                                   surf_pages=True).ensure_checked()
```
I haven't found a usage of this. It is the normal implementation of ensure_checked() that has been tested elsewhere.

prichard77 commented on cfme/cloud/provider/__init__.py:
```diff
@@ -289,7 +290,8 @@ class EditTags(CFMENavigateStep):
     def step(self, *args, **kwargs):
-        self.prerequisite_view.entities.get_entity(name=self.obj.name, surf_pages=True).check()
+        self.prerequisite_view.entities.get_entity(name=self.obj.name,
+                                                   surf_pages=True).ensure_checked()
```
I haven't found a usage of this. It is the normal implementation of ensure_checked() that has been tested elsewhere.

prichard77 commented on cfme/common/provider.py:
```diff
@@ -744,7 +744,7 @@ def refresh_provider_relationships_ui(self, from_list_view=False, wait=0, delay=
             view = navigate_to(self, 'All')
             entity = view.entities.get_entity(name=self.name, surf_pages=True)
-            entity.check()
+            entity.ensure_checked()
```
I haven't found a usage of this. It is the normal implementation of ensure_checked() that has been tested elsewhere.

prichard77 commented on cfme/tests/cloud/test_instance_power_control.py:
```diff
@@ -330,8 +330,8 @@ def _get_view_with_icons_checked():
-        view.entities.get_entity(name=testing_instance.name).check()
-        view.entities.get_entity(name=testing_instance2.name).check()
+        view.entities.get_entity(name=testing_instance.name).ensure_checked()
+        view.entities.get_entity(name=testing_instance2.name).ensure_checked()
```
This test will be part of PRT.

prichard77 commented on cfme/tests/cloud_infra_common/test_vm_instance_analysis.py:
```diff
@@ -971,7 +971,7 @@ def test_ssa_multiple_vms(ssa_multiple_vms, soft_assert, appliance, compare_linu
     view.paginator.set_items_per_page(1000)
     for ssa_vm in ssa_multiple_vms:
-        view.entities.get_entity(name=ssa_vm.name, surf_pages=True).check()
+        view.entities.get_entity(name=ssa_vm.name, surf_pages=True).ensure_checked()
```
Tested and I am adding to PRT.

prichard77 commented on cfme/tests/infrastructure/test_host.py:
```diff
@@ -90,7 +90,7 @@ def navigate_and_select_quads(provider):
     hosts_view = navigate_to(provider, 'ProviderNodes')
     assert hosts_view.is_displayed
-    [h.check() for h in hosts_view.entities.get_all()]
+    [h.ensure_checked() for h in hosts_view.entities.get_all()]
```
This has been tested with cfme/tests/infrastructure/test_host.py::test_multiple_host_good_creds and the test will be added to PRT.

prichard77 commented:
Something seems to be wrong with my PRT format. It's not running what I expect. Output doesn't tell me what is wrong , so I'll just keep trying until it works or I give up.

prichard77 commented:
I have updated the format of my PRT command a couple times. I'll get some assistance debugging this on Monday. Am marking RFR to possibly get this looked at while trying to figure out the PRT issue. I also added comments about the testing of each update.
